google/cloud/internal/credentials_impl.cc

```diff
@@ -13,25 +13,76 @@
 // limitations under the License.
 
 #include "google/cloud/internal/credentials_impl.h"
+#include "google/cloud/common_options.h"
+#include "google/cloud/internal/populate_common_options.h"
+#include <chrono>
 
 namespace google {
 namespace cloud {
 GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
 namespace internal {
+namespace {
+
+auto constexpr kDefaultScope = "https://www.googleapis.com/auth/cloud-platform";
+auto constexpr kDefaultTokenLifetime = std::chrono::hours(1);
+
+}  // namespace
+
+Options PopulateAuthOptions(Options options) {
+  // First set any defaults that may be missing.
+  options =
+      MergeOptions(std::move(options),
+                   Options{}
+                       .set<ScopesOption>({kDefaultScope})
+                       .set<AccessTokenLifetimeOption>(kDefaultTokenLifetime));
+  // Then apply any overrides.
+  return MergeOptions(
+      Options{}.set<TracingComponentsOption>(DefaultTracingComponents()),
+      std::move(options));
```

Comment:
why are these overrides?

Reply:
I think because https://github.com/googleapis/google-cloud-cpp/blob/main/doc/adr/2022-11-16-option-precedence-hierarchy.md `DefaultTracingComponents()` reads the environment variable.

Reply:
If no environment variables are set, it overrides the customer supplied options.

Maybe we should be defaulting `options` before they get here.